### ISSUE TITLE
tryUntil error: ioctl failed: No such device

### ISSUE THREAD

**Issue by leondelyon3** [F5Networks/f5-declarative-onboarding#323]
### Environment
 Declarative Onboarding Version: 1.32.0
Tenant BIG-IP Version: 15.1.6.1 deployed in one partition with two blades. Velos F5OS v1.5.0
LAG/Trunk configuration on the Velos partition, 2 ports (100Gb) used.
3 VLANS defined belonging to the same LAG/TRUNK.

### Summary
Got following error, from restnoded.log, during the DO deployment:
tryUntil error: 01070712:3: Cannot get device index, then rollback when Max tries reached
**Same DO deployment works on a BigIP VE**

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-declarative-onboarding/master/src/schema/latest/base.schema.json",
    "schemaVersion": "1.23.0",
    "class": "Device",
    "async": true,
    "label": "my BIG-IP declaration for declarative onboarding",
    "controls": {
        "class": "Controls",
        "trace": true,
        "traceResponse": true
    },
    "Common": {
        "class": "Tenant",
        "mySystem": {
            "class": "System",
            "hostname": "FI.lab",
            "cliInactivityTimeout": 0,
            "consoleInactivityTimeout": 0,
            "autoPhonehome": false
        },
        "myProvisioning": {
            "class": "Provision",
            "ltm": "nominal"
        },
        "myDns": {
            "class": "DNS",
            "nameServers": [
                "10.154.1.241",
                "10.154.1.242",
                "10.154.1.243"
            ],
            "search": [
                "localhost"
            ]
        },
        "myNtp": {
            "class": "NTP",
            "servers": [
                "10.154.1.241",
                "10.154.1.242",
                "10.154.1.243"
            ],
            "timezone": "Europe/Paris"
        },
        "Default": {
            "class": "RouteDomain",
            "id": 0,
            "strict": true,
            "vlans": [
                "http-tunnel",
                "socks-tunnel"
            ]
        },
        "RD_Gi_Internal": {
            "class": "RouteDomain",
            "id": 11,
            "routingProtocols": [
                "BGP"
            ],
            "strict": false,
            "vlans": [
                "Gi_Internal"
            ]
        },
        "UPF-self-IP": {
            "class": "SelfIp",
            "address": "172.16.1.253/24",
            "vlan": "SPGW_UPF",
            "allowService": [
                "tcp:22",
                "tcp:8443"
            ],
            "trafficGroup": "traffic-group-local-only"
        },
        "Internet-self-IP": {
            "class": "SelfIp",
            "address": "10.102.0.253/24",
            "vlan": "Internet",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "Gi-Internal-self-IP": {
            "class": "SelfIp",
            "address": "10.102.1.253%11/24",
            "vlan": "Gi_Internal",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "myDbVariables": {
            "class": "DbVariables",
            "provision.extramb": 1000,
            "restjavad.useextramb": "true",
            "users.strictpasswords": "disable",
            "ui.advisory.enabled": "true",
            "ui.advisory.color": "red",
            "ui.advisory.text": "Configuration deployed with DO/AS3. Do not make any change directly on the BIG-IP or those changes may be lost."
        }
    }
}
```

2. Observe the following error response:
```json
{
        "id": "1f79cd19-4955-4aac-9659-fcd382e568b2",
        "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/1f79cd19-4955-4aac-9659-fcd382e568b2",
        "code": 422,
        "status": "ERROR",
        "message": "invalid config - rolled back",
        "errors": [
            "tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device",
            "tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device"
        ],
        "result": {
            "class": "Result",
            "code": 422,
            "status": "ERROR",
            "dryRun": false,
            "message": "invalid config - rolled back",
            "errors": [
                "tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device",
                "tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device"
            ]
        },
        "declaration": {
            "$schema": "https://raw.githubusercontent.com/F5Networks/f5-declarative-onboarding/master/src/schema/latest/base.schema.json",
            "schemaVersion": "1.23.0",
            "class": "Device",
            "async": true,
            "label": "my BIG-IP declaration for declarative onboarding",
            "controls": {
                "class": "Controls",
                "userAgent": "vscode-f5/3.8.6",
                "dryRun": false,
                "trace": false,
                "traceResponse": false
            },
            "Common": {
                "class": "Tenant",
                "mySystem": {
                    "class": "System",
                    "hostname": "FI.lab",
                    "cliInactivityTimeout": 0,
                    "consoleInactivityTimeout": 0,
                    "autoPhonehome": false,
                    "autoCheck": true,
                    "tmshAuditLog": true,
                    "guiAuditLog": false,
                    "mcpAuditLog": "enable",
                    "preserveOrigDhcpRoutes": false,
                    "guiSecurityBanner": true,
                    "guiSecurityBannerText": "Welcome to the BIG-IP Configuration Utility.\n\nLog in with your username and password using the fields on the left."
                },
                "myProvisioning": {
                    "class": "Provision",
                    "ltm": "nominal"
                },
                "myDns": {
                    "class": "DNS",
                    "nameServers": [
                        "10.154.1.241",
                        "10.154.1.242",
                        "10.154.1.243"
                    ],
                    "search": [
                        "localhost"
                    ]
                },
                "myNtp": {
                    "class": "NTP",
                    "servers": [
                        "10.154.1.241",
                        "10.154.1.242",
                        "10.154.1.243"
                    ],
                    "timezone": "Europe/Paris"
                },
                "Default": {
                    "class": "RouteDomain",
                    "id": 0,
                    "strict": true,
                    "vlans": [
                        "http-tunnel",
                        "socks-tunnel"
                    ],
                    "connectionLimit": 0
                },
                "RD_Gi_Internal": {
                    "class": "RouteDomain",
                    "id": 11,
                    "routingProtocols": [
                        "BGP"
                    ],
                    "strict": false,
                    "vlans": [
                        "Gi_Internal"
                    ],
                    "connectionLimit": 0
                },
                "UPF-self-IP": {
                    "class": "SelfIp",
                    "address": "172.16.1.253/24",
                    "vlan": "SPGW_UPF",
                    "allowService": [
                        "tcp:22",
                        "tcp:8443"
                    ],
                    "trafficGroup": "traffic-group-local-only"
                },
                "Internet-self-IP": {
                    "class": "SelfIp",
                    "address": "10.102.0.253/24",
                    "vlan": "Internet",
                    "allowService": "default",
                    "trafficGroup": "traffic-group-local-only"
                },
                "Gi-Internal-self-IP": {
                    "class": "SelfIp",
                    "address": "10.102.1.253%11/24",
                    "vlan": "Gi_Internal",
                    "allowService": "default",
                    "trafficGroup": "traffic-group-local-only"
                },
                "myDbVariables": {
                    "class": "DbVariables",
                    "provision.extramb": 1000,
                    "restjavad.useextramb": "true",
                    "ui.advisory.enabled": "true",
                    "ui.advisory.color": "red",
                    "ui.advisory.text": "Configuration deployed with DO/AS3. Do not make any change directly on the BIG-IP or those changes may be lost."
                }
            }
        },
        "httpStatus": 200
    }
```
The restnoded.log:
Thu, 08 Sep 2022 08:14:48 GMT - finest: socket 202 opened
Thu, 08 Sep 2022 08:14:48 GMT - fine: [f5-declarative-onboarding: restWorker.js] create localhost /tm/net/self {"vlan":"/Common/Gi_Internal","name":"Gi-Internal-self-IP","partition":"Common","address":"10.102.1.253%11/24","trafficGroup":"traffic-group-local-only","allowService":"default"}
Thu, 08 Sep 2022 08:14:48 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":400}
Thu, 08 Sep 2022 08:14:48 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
Thu, 08 Sep 2022 08:14:48 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device tries left: 21
Thu, 08 Sep 2022 08:14:48 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 21 , code: 400 , message: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device
Thu, 08 Sep 2022 08:14:50 GMT - fine: [f5-declarative-onboarding: restWorker.js] create localhost /tm/net/self {"vlan":"/Common/Gi_Internal","name":"Gi-Internal-self-IP","partition":"Common","address":"10.102.1.253%11/24","trafficGroup":"traffic-group-local-only","allowService":"default"}
Thu, 08 Sep 2022 08:14:51 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":400}
Thu, 08 Sep 2022 08:14:51 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
Thu, 08 Sep 2022 08:14:51 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device tries left: 20
Thu, 08 Sep 2022 08:14:51 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 20 , code: 400 , message: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device

Thu, 08 Sep 2022 08:15:31 GMT - finer: [f5-declarative-onboarding: restWorker.js] Max tries reached.
Thu, 08 Sep 2022 08:15:31 GMT - severe: [f5-declarative-onboarding: networkHandler.js] Error creating self IPs: tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device
Thu, 08 Sep 2022 08:15:31 GMT - severe: [f5-declarative-onboarding: networkHandler.js] Error processing network declaration: tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device
Thu, 08 Sep 2022 08:15:31 GMT - severe: [f5-declarative-onboarding: declarationHandler.js] Error processing declaration: tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device
Thu, 08 Sep 2022 08:15:31 GMT - severe: [f5-declarative-onboarding: restWorker.js] Error onboarding: tryUntil: max tries reached: 01070712:3: Cannot get device index for Gi_Internal in rd11 - ioctl failed: No such device
Thu, 08 Sep 2022 08:15:31 GMT - info: [f5-declarative-onboarding: restWorker.js] Rolling back configuration

### Expected Behavior
The same DO deployment succeed in BigIP VE:
```json

    "id": "a10ee956-4d8d-4b93-af01-edf17d53e649",
    "selfLink": "https://localhost/mgmt/shared/declarative-onboarding/task/a10ee956-4d8d-4b93-af01-edf17d53e649",
    "result": {
        "class": "Result",
        "code": 200,
        "status": "OK",
        "dryRun": false,
        "message": "success"
    },
    "declaration": {
        "$schema": "https://raw.githubusercontent.com/F5Networks/f5-declarative-onboarding/master/src/schema/latest/base.schema.json",
        "schemaVersion": "1.23.0",
        "class": "Device",
        "async": true,
        "label": "my BIG-IP declaration for declarative onboarding",
        "controls": {
            "class": "Controls",
            "userAgent": "vscode-f5/3.8.6",
            "dryRun": false,
            "trace": false,
            "traceResponse": false
        },
        "Common": {
            "class": "Tenant",
            "mySystem": {
                "class": "System",
                "hostname": "FI.lab",
                "cliInactivityTimeout": 0,
                "consoleInactivityTimeout": 0,
                "autoPhonehome": false,
                "autoCheck": true,
                "tmshAuditLog": true,
                "guiAuditLog": false,
                "mcpAuditLog": "enable",
                "preserveOrigDhcpRoutes": false,
                "guiSecurityBanner": true,
                "guiSecurityBannerText": "Welcome to the BIG-IP Configuration Utility.\n\nLog in with your username and password using the fields on the left."
            },
            "myProvisioning": {
                "class": "Provision",
                "ltm": "nominal"
            },
            "myDns": {
                "class": "DNS",
                "nameServers": [
                    "10.154.1.241",
                    "10.154.1.242",
                    "10.154.1.243"
                ],
                "search": [
                    "localhost"
                ]
            },
            "myNtp": {
                "class": "NTP",
                "servers": [
                    "10.154.1.241",
                    "10.154.1.242",
                    "10.154.1.243"
                ],
                "timezone": "Europe/Paris"
            },
            "Default": {
                "class": "RouteDomain",
                "id": 0,
                "strict": true,
                "vlans": [
                    "http-tunnel",
                    "socks-tunnel"
                ],
                "connectionLimit": 0
            },
            "RD_Gi_Internal": {
                "class": "RouteDomain",
                "id": 11,
                "routingProtocols": [
                    "BGP"
                ],
                "strict": false,
                "vlans": [
                    "Gi_Internal"
                ],
                "connectionLimit": 0
            },
            "UPF-self-IP": {
                "class": "SelfIp",
                "address": "172.16.1.253/24",
                "vlan": "SPGW_UPF",
                "allowService": [
                    "tcp:22",
                    "tcp:8443"
                ],
                "trafficGroup": "traffic-group-local-only"
            },
            "Internet-self-IP": {
                "class": "SelfIp",
                "address": "10.102.0.253/24",
                "vlan": "Internet",
                "allowService": "default",
                "trafficGroup": "traffic-group-local-only"
            },
            "Gi-Internal-self-IP": {
                "class": "SelfIp",
                "address": "10.102.1.253%11/24",
                "vlan": "Gi_Internal",
                "allowService": "default",
                "trafficGroup": "traffic-group-local-only"
            },
            "myDbVariables": {
                "class": "DbVariables",
                "provision.extramb": 1000,
                "restjavad.useextramb": "true",
                "ui.advisory.enabled": "true",
                "ui.advisory.color": "red",
                "ui.advisory.text": "Configuration deployed with DO/AS3. Do not make any change directly on the BIG-IP or those changes may be lost."
            }
        }
    }
}
```

### Actual Behavior
DO deployment failed, then rollback.

**Comment by sunitharonan** [comment]
Hi @leondelyon3, please contact us at automation_toolchain_pm@f5.com so we can understand more and prioritize.

**Comment by mdditt2000** [comment]
Customer Identified and prioritized

**Comment by sunitharonan** [comment]
This issue has been resolved in Do 1.35.0
